test/integration/use_test.rb

```diff
@@ -1,13 +1,69 @@
 require 'test_helper'
+require 'rack/test'
 
 describe 'Rack middleware integration' do
+  include Rack::Test::Methods
+
+  def response
+    last_response
+  end
+
   describe '.use' do
+    let(:app) { UseActionApplication }
+
     it 'uses the specified Rack middleware' do
-      response = Rack::MockRequest.new(UseAction.new).get('/')
-      status, headers, _ = response
+      router = Lotus::Router.new do
+        get '/', to: 'use_action#index'
+      end
+
+      UseActionApplication = Rack::Builder.new do
+        run router
+      end.to_app
+
+      get '/'
+
+      UseAction::Index.respond_to?(:call).must_equal true
+      response.status.must_equal 200
+      response.headers.fetch('X-Middleware').must_equal 'OK'
+    end
+  end
+
+  describe 'using .use more than one time' do
+    let(:app) { DoubleUseActionApplication }
+
+    it 'rack builder is instantiate only one time' do
+      router = Lotus::Router.new do
+        get '/',     to: 'use_action#index'
+        get '/show', to: 'use_action#show'
+      end
+
+      DoubleUseActionApplication = Rack::Builder.new do
+        run router
+      end.to_app
+
+      get '/'
+
+      UseAction::Index.rack_builder == UseAction::Show.rack_builder
```

Comment:
I don't think this would add a value. We're testing a detail, instead of a behavior. Also, if Rack Team customizes `Rack::Builder#==` this test will fail in the future.

The behavior that we want to describe is: given two actions with separated middleware stacks, those shall not to interfers each other.

What if we introduce `YMiddleware` and let `UseAction::Show` to use it? Then we could introspect the response and headers returned from `/show`.